src/test/java/ch/jalu/configme/SettingsManagerBuilderTest.java

```diff
@@ -156,4 +161,142 @@ void shouldCreateSettingsManagerFromFileObject2() throws URISyntaxException {
+    }
+
+    /**
+     * This method tests the {@link VersionMigrationService} class.
+     * @author gamerover98
+     */
+    @Test
+    void shouldNotMigrateAndKeepConfigValues() throws IOException {
+        // given
+        Path file = copyFileFromResources("/versions/config-current-version-sample.yml", temporaryFolder);
+        long initialFileSize = Files.size(file);
+
+        ConfigurationData configurationData = ConfigurationDataBuilder.createConfiguration(TestVersionConfiguration.class);
+        MigrationService migrationService = getVersionMigrationService();
+
+        // when
+        SettingsManagerImpl manager = (SettingsManagerImpl) SettingsManagerBuilder.withYamlFile(file)
+            .configurationData(configurationData)
+            .migrationService(migrationService)
+            .create();
+
+        // then
+        // the file won't change.
+        assertThat(Files.size(file), equalTo(initialFileSize));
+
+        PropertyReader reader = manager.getPropertyResource().createReader();
+
+        assertThat(TestVersionConfiguration.VERSION_NUMBER.determineValue(reader).getValue(), equalTo(2));
+        assertThat(TestVersionConfiguration.SHELF_POPATOES.determineValue(reader).getValue(), equalTo(4));
+        assertThat(TestVersionConfiguration.SHELF_TOMATOES.determineValue(reader).getValue(), equalTo(10));
+    }
+
+    /**
+     * This method tests the {@link VersionMigrationService} class.
+     * @author gamerover98
+     */
+    @Test
+    void shouldNotMigrateTheNextVersionAndThrowException() {
+        // given
+        Path file = copyFileFromResources("/versions/config-next-version-sample.yml", temporaryFolder);
+
+        ConfigurationData configurationData = ConfigurationDataBuilder.createConfiguration(TestVersionConfiguration.class);
+        MigrationService migrationService = getVersionMigrationService();
+
+        // throws an illegal state because the config version is greater than the current version.
+        assertThrows(
+            IllegalStateException.class,
+            () -> SettingsManagerBuilder.withYamlFile(file)
+                .configurationData(configurationData)
+                .migrationService(migrationService)
+                .create());
+    }
+
+    /**
+     * This method tests the {@link VersionMigrationService} class.
+     * @author gamerover98
+     */
+    @Test
+    void shouldNotMigrateFromAnInvalidVersionAndThrowException() {
+        // given
+        Path file = copyFileFromResources("/versions/config-invalid-version-sample.yml", temporaryFolder);
+
+        ConfigurationData configurationData = ConfigurationDataBuilder.createConfiguration(TestVersionConfiguration.class);
+        MigrationService migrationService = getVersionMigrationService();
+
+        // throws an illegal state because the config version is lower than the start version.
+        assertThrows(
+            IllegalStateException.class,
```

Comment:
I think one test that goes via SettingsManagerBuilder and everything is fine, but since this test here and others test VersionMigrationService stuff, I would create a new class VersionMigrationServiceTest with tests that just use VersionMigrationService itself